src/System.Net.Http/tests/FunctionalTests/IdnaProtocolTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Net.Test.Common;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public abstract class IdnaProtocolTests : HttpClientTestBase
+    {
+        protected abstract bool SupportsIdna { get; }
```

Comment:
Nit: you could make it virtual with => true and then not have to override it for true later. That's generally what's done elsewhere.